packages/rum-core/src/domain/rumEventsCollection/action/getActionNameFromElement.ts

```diff
@@ -1,36 +1,39 @@
 import { safeTruncate } from '@datadog/browser-core'
 
-export function getActionNameFromElement(element: Element): string {
+/**
+ * Get the action name from the attribute 'data-dd-action-name' on the element or any of its parent.
+ * It can also be retrieved from a user defined attribute.
+ */
+const DEFAULT_PROGRAMMATIC_ATTRIBUTE = 'data-dd-action-name'
+
+export function getActionNameFromElement(element: Element, userProgrammaticAttribute?: string): string {
   // Proceed to get the action name in two steps:
   // * first, get the name programmatically, explicitly defined by the user.
   // * then, use strategies that are known to return good results. Those strategies will be used on
   //   the element and a few parents, but it's likely that they won't succeed at all.
   // * if no name is found this way, use strategies returning less accurate names as a fallback.
   //   Those are much likely to succeed.
   return (
-    getActionNameFromElementProgrammatically(element) ||
+    getActionNameFromElementProgrammatically(element, DEFAULT_PROGRAMMATIC_ATTRIBUTE) ||
+    (userProgrammaticAttribute && getActionNameFromElementProgrammatically(element, userProgrammaticAttribute)) ||
```

Comment:
Can you please explain why the default attribute has priority over the user custom attribute?

Reply:
it seemed more convenient for reusing a previous tagging but allow to override it with our own attribute if needed